ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/schema/PipelineMetadataStore.java

```diff
@@ -0,0 +1,134 @@
+package com.mozilla.telemetry.ingestion.core.schema;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.auto.value.AutoValue;
+import com.mozilla.telemetry.ingestion.core.util.IOFunction;
+import com.mozilla.telemetry.ingestion.core.util.Json;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * This schema store uses the same mozilla-pipeline-schemas tarball as JSONSchemaStore,
+ * but instead of returning Schema instances to be used for validation, it returns
+ * {@link PipelineMetadata} objects parsed from each schema's "mozPipelineMetadata" entry.
+ */
+public class PipelineMetadataStore extends SchemaStore<PipelineMetadataStore.PipelineMetadata> {
+
+  @AutoValue
+  @JsonDeserialize(builder = AutoValue_PipelineMetadataStore_JweMapping.Builder.class)
```

Comment:
Everywhere else in this repo, we interact with JSON via `JsonNode` and the tree interface, but this `PipelineMetadataStore` uses Jackson databind to instantiate a class defined with AutoValue. It's definitely cleaner to work with this structure, since all the parsing happens centrally. But, it's adding a new concept to the codebase, so I'm open to feedback there.

This structure is particularly convenient because it handles constructing concrete `JsonPointer` instances as part of the cached PipelineMetadata objects, whereas we would likely have to reconstruct those every time or maintain a separate cache if we returned an `ObjectNode` here.

Reply:
This effectively turns the json node into an object with known fields at compile time?

Reply:
> This effectively turns the json node into an object with known fields at compile time?

Exactly. Using Jackson databind is essentially a validation step and a construction step. Jackson will throw an error if it finds incompatible types between what's present in the JSON and what's defined in the class fields, for example.